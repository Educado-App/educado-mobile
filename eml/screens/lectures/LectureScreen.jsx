import React, { useState, useEffect } from 'react';
import { View } from 'react-native';
import VideoLectureScreen from './VideoLectureScreen';
import TextImageLectureScreen from './TextImageLectureScreen';
import PropTypes from 'prop-types';
import Text from '../../components/general/Text';

export default function LectureScreen({ lectureObject, courseObject, currentIndex, indexCount }) {

	const [course, setCourse] = useState(courseObject);
	const [lecture, setLecture] = useState(lectureObject);
	const isLastSlide = currentIndex === indexCount - 1;
  
	useEffect(() => {
		setLecture(lectureObject);
		setCourse(courseObject);
	}, []);


	//Safe area should not be used if we want to use the full screen
	return (
		<View className="flex-1 bg-projectWhite">

			{lecture && course ?
				<View className="w-full h-full flex-col justify-center items-center">

					{lecture.video ?
						<VideoLectureScreen lectureObject={lecture} courseObject={course} isLastSlide={isLastSlide} />
						:
						<TextImageLectureScreen lectureObject={lecture} courseObject={course} isLastSlide={isLastSlide} />
					}
          
				</View>
				:
				<View className="w-full h-full items-center justify-center align-middle">
					<Text className="text-[25px] font-bold ml-[10]">loading...</Text>
				</View>

			}
		</View>
	);
}

LectureScreen.propTypes = {
	lectureObject: PropTypes.object,
	courseObject: PropTypes.object,
	currentIndex: PropTypes.number,
	indexCount: PropTypes.number,
};