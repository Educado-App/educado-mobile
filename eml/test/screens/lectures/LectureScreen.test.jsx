import React from 'react';
import { render } from '@testing-library/react-native';
import { useNavigation } from '@react-navigation/native';
import LectureScreen from '../../../screens/lectures/LectureScreen';

// Mock the useNavigation hook
jest.mock('@react-navigation/native', () => ({
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: jest.fn(),
}));

describe('LectureScreen', () => {
    it('shows loading when lecture and course are not available', () => {
        // Setup mock
        useNavigation.mockReturnValue({ goBack: jest.fn() });

        const route = {
            params: {
                lectureId: 'mockLectureId',
                courseId: 'mockCourseId'
            }
        };

        const { getByText } = render(<LectureScreen route={route} />);
        expect(getByText('loading...')).toBeTruthy();
    });

    // Add more test cases...
});
