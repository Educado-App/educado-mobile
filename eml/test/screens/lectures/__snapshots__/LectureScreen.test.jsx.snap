// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LectureScreen renders CourseScreen with courses loaded 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "width": "100%",
        },
        {
          "height": "100%",
        },
        {
          "alignItems": "center",
        },
        {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 25,
          },
          {
            "fontWeight": "700",
          },
          {
            "marginLeft": 10,
          },
        ]
      }
    >
      loading...
    </Text>
  </View>
</View>
`;

exports[`LectureScreen renders LectureScreen correctly when there are no lecture 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "width": "100%",
        },
        {
          "height": "100%",
        },
        {
          "alignItems": "center",
        },
        {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 25,
          },
          {
            "fontWeight": "700",
          },
          {
            "marginLeft": 10,
          },
        ]
      }
    >
      loading...
    </Text>
  </View>
</View>
`;
